homeassistant/components/spotify/config_flow.py

```diff
@@ -34,27 +36,25 @@ def extra_authorize_data(self) -> dict[str, Any]:
 
     async def async_oauth_create_entry(self, data: dict[str, Any]) -> ConfigFlowResult:
         """Create an entry for Spotify."""
-        spotify = Spotify(auth=data["token"]["access_token"])
+        spotify = SpotifyClient(async_get_clientsession(self.hass))
+        spotify.authenticate(data[CONF_TOKEN][CONF_ACCESS_TOKEN])
 
         try:
-            current_user = await self.hass.async_add_executor_job(spotify.current_user)
+            current_user = await spotify.get_current_user()
         except Exception:  # noqa: BLE001
+            self.logger.exception("Failed to fetch user details")
```

Comment:
Don't we have better control over exceptions with the new library?

Reply:
We have, but I wanted to add the extra exceptions in a follow up, since they also add extra tests